data/json/requirements/materials.json

```diff
@@ -12,6 +12,15 @@
     "components": [ [ [ "lead", 1 ], [ "bismuth", 1 ], [ "silver_small", 1 ], [ "gold_small", 1 ] ] ]
   },
   {
+    "id": "ammo_bullet",
```

Comment:
It's encouraging to see you trying out new approaches for recipe layout but this is probably more complexity than it's worth. You can just specify `ammo_bullet` and then `copper` separately as there aren't any alternatives to the latter.

The first step in this PR is to actually define `copper` in `metals.json` with an approximately correct density (see `lead`) and update any non-ammo recipes first.